app/views.py

```diff
@@ -69,35 +89,90 @@ def editProjectView(request, projectID):
         if form.is_valid():
             newData = form.cleaned_data
             newData["id"] = projectID
-            message, status_code = ProjectsAPIView.updateProject(newData)
 
-            if status_code != status.HTTP_200_OK:
+            # Replace user emails with userIDs to save
+
+            # Splits on comma, space, or newline. Makes sure only unique emails
+            emails = set(re.split(r"\s*[, \n]+\s*", newData["contributors"].strip()))
+            del newData["contributors"]
+            emails = [{"email": email} for email in emails if email]
```

Comment:
there should probably be some check to prevent a user from adding their own email or an existing contributor (if it's not already included)

Reply:
This is implemented because in python, sets don't allow duplicates. Also, it will still work if user's add their own email (see lines 111-112).